warehouse/query-core/src/main/java/datawave/query/tables/async/event/VisitorFunction.java

```diff
@@ -215,6 +217,65 @@ public ScannerChunk apply(@Nullable ScannerChunk input) {
+                            // copy the config and set minimum thresholds for collapsing Ors
+                            ShardQueryConfiguration reduceConfig = new ShardQueryConfiguration(config);
+                            reduceConfig.setMaxOrExpansionThreshold(2);
+                            reduceConfig.setMaxOrRangeThreshold(2);
+                            
+                            try {
+                                Query settings = config.getQuery();
+                                
+                                if (config.canHandleExceededValueThreshold()) {
+                                    URI hdfsQueryCacheUri = getFstHdfsQueryCacheUri(config, settings);
+                                    SortedMap<Integer,List<String>> pushdownCapacity;
+                                    FileSystem fs = null;
+                                    String fstHdfsUri = null;
+                                    if (hdfsQueryCacheUri != null) {
+                                        fs = VisitorFunction.fileSystemCache.getFileSystem(hdfsQueryCacheUri);
+                                        fstHdfsUri = hdfsQueryCacheUri.toString();
+                                    }
+                                    
+                                    pushdownCapacity = PushdownLargeFieldedListsVisitor.getPushdownCapacity(reduceConfig, script, fs, fstHdfsUri);
+                                    
+                                    // determine if its possible to reduce enough to meet the threshold
+                                    int capacitySum = 0;
+                                    for (Integer capacity : pushdownCapacity.keySet()) {
+                                        capacitySum += capacity * pushdownCapacity.get(capacity).size();
+                                    }
+                                    
+                                    if (termCount - capacitySum <= config.getMaxTermThreshold()) {
+                                        // sort from largest to smallest reductions and make reductions until under the threshold
+                                        Set<String> fieldsToReduce = new HashSet<>();
+                                        int toReduce = termCount - config.getMaxTermThreshold();
+                                        while (toReduce > 0) {
+                                            // get the highest value field out of the map
+                                            Integer reduction = pushdownCapacity.lastKey();
+                                            List<String> fields = pushdownCapacity.get(reduction);
+                                            
+                                            // take the first field
+                                            String field = fields.remove(0);
+                                            fieldsToReduce.add(field);
+                                            toReduce -= reduction;
```

Comment:
If we're counting the list ivarator nodes as a single term, shouldn't this be something like:
```
toReduce -= (reduction-1);
```